x-pack/elastic-agent/pkg/agent/cmd/run.go

```diff
@@ -44,6 +47,10 @@ const (
 	agentName = "elastic-agent"
 )
 
+func init() {
+	apm.DefaultTracer.Close()
```

Comment:
> as apm.DefaultTracer should be closed due to the fact a new tracer is create by NewTracer or NewTracerOptions, better to have them both together. It'll make this relation more explicitly .

The default tracer is not closed because a new one is being created, it's being closed because the default one is not ever expected to be used by Elastic Agent. If it is left open, it will periodically attempt to connect to APM Server. Closing it in package init prevents that as early as possible.

> If in the far future someone else changes the code and decides to use the default tracer, there is a good change they will miss it's being closed here.

We will be releasing a new major version of the agent soon (weeks), and this new version has a more sensible API which does not have side effects due to importing the agent's main package:

```go
package apm

// DefaultTracer returns the default global Tracer, set the first time the
// function is called. It is configured via environment variables.
//
// This will always be initialized to a non-nil value. If any of the
// environment variables are invalid, the corresponding errors will be logged
// to stderr and the default values will be used instead.
func DefaultTracer() *Tracer

// SetDefaultTracer sets the tracer returned by DefaultTracer(). If another
// tracer has already been initialized, it is closed. Any queued events are not
// flushed; it is the responsibility of the caller to call
// DefaultTracer().Flush()
```

I suggest we stick with closing the tracer in package-init for now, and migrate to the new version of the agent ASAP. Then the close call can simply be removed.